client/sites/components/sites-dataviews/actions.tsx

```diff
@@ -13,19 +15,90 @@ import {
 	isNotAtomicJetpack,
 	isP2Site,
 	isSimpleSite,
+	isDisconnectedJetpackAndNotAtomic,
 } from 'calypso/sites-dashboard/utils';
-import { useDispatch as useReduxDispatch } from 'calypso/state';
+import { useDispatch as useReduxDispatch, useSelector } from 'calypso/state';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { launchSiteOrRedirectToLaunchSignupFlow } from 'calypso/state/sites/launch/actions';
 import type { SiteExcerptData } from '@automattic/sites';
 import type { Action } from '@wordpress/dataviews';
-
-export function useActions(): Action< SiteExcerptData >[] {
+export function useActions( {
+	openSitePreviewPane,
+	selectedItem,
+}: {
+	openSitePreviewPane?: (
+		site: SiteExcerptData,
+		source: 'site_field' | 'action' | 'list_row_click' | 'environment_switcher'
+	) => void;
+	selectedItem?: SiteExcerptData | null;
+} ): Action< SiteExcerptData >[] {
 	const { __ } = useI18n();
 	const dispatch = useReduxDispatch();
 
+	const capabilities = useSelector<
+		{
+			currentUser: {
+				capabilities: Record< string, Record< string, boolean > >;
+			};
+		},
+		Record< string, Record< string, boolean > >
+	>( ( state ) => state.currentUser.capabilities );
+
 	return useMemo(
 		() => [
+			{
+				id: 'site-overview',
+				isPrimary: true,
+				label: __( 'Overview' ),
+				icon: drawerLeft,
+				callback: ( sites ) => {
+					const site = sites[ 0 ];
+					const adminUrl = site.options?.admin_url ?? '';
+					const isAdmin = capabilities[ site.ID ]?.manage_options;
+					if (
+						isAdmin &&
+						! isP2Site( site ) &&
+						! isNotAtomicJetpack( site ) &&
+						! isDisconnectedJetpackAndNotAtomic( site )
+					) {
+						openSitePreviewPane && openSitePreviewPane( site, 'action' );
+					} else {
+						navigate( adminUrl );
+					}
+				},
+				isEligible: ( site ) => {
+					if ( site.ID === selectedItem?.ID ) {
```

Comment:
Judging by the code, I presume this is for hiding the `site-overview` action when the item is selected (list view). This works, but it also runs the logic for every item. An alternative would be to pass the `view.type` to the useActions hook and only add `site-overview` to the actions array if the view is table.

Reply:
@okmttdhr actually, this is what happens (note how the actions changes after clicking):

https://github.com/user-attachments/assets/8afc1fb2-ef05-4a4a-9e2b-a81113e8073b

Reply:
Thank you for your review!

I agree that hiding the Overview primary action in the list view makes sense—it ensures consistency, and the row-click interaction is provided for the list view. I'll spin up a PR to address this.